app/controllers/pets_controller.rb

```diff
@@ -24,9 +24,16 @@ def edit
 
   def update
     pet = Pet.find(params[:id])
-    pet.update(pet_params)
-
-    redirect_to "/pets/#{pet.id}"
+    if pet.update(pet_params)
+      redirect_to "/pets/#{pet.id}"
+    else
+      flash[:error] = ""
+      redirect_back(fallback_location: root_path)
+      missing(pet_params).each do |param|
+        flash[:error] += "The #{param} field must not be left blank. "
+      end
+      flash[:error]
+    end
```

Comment:
this is the same code logic for the shelter editing. if we have time this would make a good partial